internal/metrics/metrics.go

```diff
@@ -0,0 +1,168 @@
+package metrics
+
+import (
+	"errors"
+	"github.com/Kopleman/metcol/internal/common"
+	"github.com/Kopleman/metcol/internal/server/store"
+	"strconv"
+	"strings"
+)
+
+type IMetrics interface {
+	SetMetric(metricType common.MetricType, name string, value string) error
+	GetValueAsString(metricType common.MetricType, name string) (string, error)
+	GetAllValuesAsString() (map[string]string, error)
+}
+
+func (m *Metrics) buildStoreKey(name string, metricType common.MetricType) string {
+	return name + "-" + string(metricType)
+}
+
+func (m *Metrics) parseStoreKey(key string) (string, common.MetricType, error) {
+	parts := strings.Split(key, "-")
+	if len(parts) != 2 {
+		return "", common.UnknownMetricType, ErrStoreKeyParse
+	}
+
+	typeAsString := parts[1]
+	switch typeAsString {
+	case string(common.CounterMetricType):
+		return parts[0], common.CounterMetricType, nil
+	case string(common.GougeMetricType):
+		return parts[0], common.GougeMetricType, nil
+	default:
+		return parts[0], common.UnknownMetricType, ErrUnknownMetricType
+	}
+}
+
+func (m *Metrics) SetGauge(name string, value float64) error {
+	storeKey := m.buildStoreKey(name, common.GougeMetricType)
+
+	_, err := m.store.Read(storeKey)
+
+	if err != nil && !errors.Is(err, store.ErrNotFound) {
+		return err
+	}
+
+	if err != nil && errors.Is(err, store.ErrNotFound) {
+		return m.store.Create(storeKey, value)
+	}
+
+	return m.store.Update(storeKey, value)
+}
+
+func (m *Metrics) SetCounter(name string, value int64) error {
+	storeKey := m.buildStoreKey(name, common.CounterMetricType)
+
+	counterValue, err := m.store.Read(storeKey)
+
+	if err != nil && !errors.Is(err, store.ErrNotFound) {
```

Comment:
Так код будет проще читать:

```suggestion
	if err != nil {
	    if errors.Is(err, store.ErrNotFound) {
	        ...
	        return
	    }
	    ...
	    return
	}
```